From 721f4e2dbe88ab01bb0e1a12cac02190f143f8ae Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 12 Jan 2021 15:44:35 -0800
Subject: [PATCH 1/2] hss-payload-generator: Rename EM_ARC_COMPACT2 to EM_ARCV2

Define a fallback to use EM_ARC_COMPACT2 if EM_ARCV2 is not defined
which could be the case if elfutils < 0.182

Upstream-Status: Submitted [https://github.com/polarfire-soc/hart-software-services/pull/10]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 tools/hss-payload-generator/elf_strings.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tools/hss-payload-generator/elf_strings.c b/tools/hss-payload-generator/elf_strings.c
index a18688f..2070369 100644
--- a/tools/hss-payload-generator/elf_strings.c
+++ b/tools/hss-payload-generator/elf_strings.c
@@ -32,6 +32,11 @@
 
 #define ARRAY_SIZE(x) (sizeof(x)/sizeof(x[0]))
 
+/* elfutils-0.182 has rename EM_ARC_COMPACT2 to EM_ARCV2 */
+#ifndef EM_ARCV2
+#define EM_ARCV2 EM_ARC_COMPACT2
+#endif
+
 static char const * const ClassStrings[] = {
 	[ELFCLASSNONE] =	"Invalid class",
 	[ELFCLASS32]  =		"32-bit objects",
@@ -216,7 +221,7 @@ static char const * const MachineStrings[] = {
 	[EM_CLOUDSHIELD] =	"CloudShield architecture family",
 	[EM_COREA_1ST] =	"KIPO-KAIST Core-A 1st generation processor family",
 	[EM_COREA_2ND] =	"KIPO-KAIST Core-A 2nd generation processor family",
-	[EM_ARC_COMPACT2] =	"Synopsys ARCompact V2",
+	[EM_ARCV2] =		"Synopsys ARCompact V2",
 	[EM_OPEN8] =		"Open8 8-bit RISC soft processor core",
 	[EM_RL78] =		"Renesas RL78 family",
 	[EM_VIDEOCORE5] =	"Broadcom VideoCore V processor",
-- 
2.30.0

